src/lib.rs

```diff
@@ -68,8 +70,90 @@ impl Default for Facility {
     }
 }
 
+impl fmt::Display for Facility {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let s = match self {
+            Facility::Kern => "Kern",
+            Facility::User => "User",
+            Facility::Mail => "Mail",
+            Facility::Daemon => "Daemon",
+            Facility::Auth => "Auth",
+            Facility::Syslog => "Syslog",
+            Facility::Lpr => "Lpr",
+            Facility::News => "News",
+            Facility::Uucp => "Uucp",
+            Facility::Cron => "Cron",
+            Facility::Authpriv => "Authpriv",
+            Facility::Ftp => "Ftp",
+            Facility::Local0 => "Local0",
+            Facility::Local1 => "Local1",
+            Facility::Local2 => "Local2",
+            Facility::Local3 => "Local3",
+            Facility::Local4 => "Local4",
+            Facility::Local5 => "Local5",
+            Facility::Local6 => "Local6",
+            Facility::Local7 => "Local7",
+        };
+
+        f.write_str(s)
+    }
+}
+
+impl<T> fmt::Display for IntToEnumError<T> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let enum_name: &'static str = std::any::type_name::<T>();
+        write!(f, "Failed to convert {} to {}", self.value, enum_name)
+    }
+}
+
+impl TryFrom<u8> for Facility {
+    type Error = IntToEnumError<Self>;
+
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        Into::<i32>::into(value).try_into()
+    }
+}
+
+/// Try convert a i32 (libc::c_int) into a Facility
+impl TryFrom<i32> for Facility {
+    type Error = IntToEnumError<Self>;
+
+    fn try_from(value: i32) -> Result<Self, Self::Error> {
+        let variant = match value {
+            0 => Self::Kern,
+            1 => Self::User,
+            2 => Self::Mail,
+            3 => Self::Daemon,
+            4 => Self::Auth,
+            5 => Self::Syslog,
+            6 => Self::Lpr,
+            7 => Self::News,
+            8 => Self::Uucp,
+            9 => Self::Cron,
+            10 => Self::Authpriv,
+            11 => Self::Ftp,
```

Comment:
Skipping 12 till up 15 is probably intended

Reply:
It sure is. I don't provide variants for NTP subsystem etc